### ISSUE TITLE
I have ran your code, but the results are not good as yours.

### ISSUE THREAD

**Issue by ZhuPeipei-CUHKSZ-TJUTCM** [jiwoon-ahn/irn#26]
I have ran your code, but the results are not good as yours. So do you have some special skills to run the code? Thanks.

Instance segmentation + training dataset (0.5AP): mine 35.7, yours 37.7;
Semantic segmentation + training dataset (miou): mine 66.0, yours 66.5;

**Comment by xiaaoo-zz** [comment]
Same here, 35.7 for instance segmentation and 65.9 for semantic segmentation.

**Comment by jiwoon-ahn** [comment]
Hi @username123062, @xiaaoo,
Detectron (https://github.com/facebookresearch/Detectron) has been changed since I finished the paper. I suppose that might have caused the performance gaps.
For semantic segmentation, we used dense CRF and Multi-scale & Flip inference. Please see the details in the supplementary materials of our paper.

**Comment by ZhuPeipei-CUHKSZ-TJUTCM** [comment]
So，could you upload the latest code to github？Thanks------------------ 原始邮件 ------------------
发件人: "Jiwoon Ahn"<notifications@github.com>
发送时间: 2020年3月3日(星期二) 下午2:41
收件人: "jiwoon-ahn/irn"<irn@noreply.github.com>;
抄送: "Peipei"<1213490284@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [jiwoon-ahn/irn] I have ran your code, but the results are notgood as yours. (#26)



Hi @username123062, @xiaaoo,
 Detectron (https://github.com/facebookresearch/Detectron) has been changed since I finished the paper. I suppose that might have caused the performance gaps.
 For semantic segmentation, we used dense CRF and Multi-scale & Flip inference. Please see the details in the supplementary materials of our paper.
 
&mdash;

**Comment by jiwoon-ahn** [comment]
I think I misunderstood your question.
If you find it difficult to reproduce our results with DeepLab, try with one of the old versions of Detectron.
But, it seems the question is about the quality of your pseudo labels. You could try the entire process several times, and see if the gap continues.
I am afraid to say that I do not know exactly why this happens. Some succeeds, and some fails. Perhaps I made some glitches while refactoring the code?

**Comment by jakecastelli** [comment]
hi guys👋  

@jiwoon-ahn thanks for the great work.

Have you tried to validate the results again lately? Any suggestions?
Cheers

**Comment by YB17** [comment]
@username123062 @xiaaoo hello friends, could you please share the hyperparameters for training the IRNet please? I tried the default parameters in run_sample.py, but the CAM is really poor and I have to lower the learning rate and increase epochs to reach a good convergence. For the IRNet the default parameters also won't work, as the displacement field can not be learned, which leads to very poor performance for the centroid clustering.  It is a pity that the original paper didn't even mention the hyperparameters, and it seems that the authors have already abandoned answering the questions posted here.  Thanks a lot!!!
